docs/dg/dev/troubleshooting/troubleshooting-performance-issues/redis-session-lock-excpetions.md

```diff
@@ -1,18 +1,47 @@
 ---
-title: Redis session lock exceptions
-description: Redis Session Lock Exceptions
+title: Redis session lock
+description: Redis Session Lock
 template: troubleshooting-guide-template
 redirect_from:
 - /docs/scos/dev/troubleshooting/troubleshooting-performance-issues/redis-session-lock.html
-last_updated: Aug 1, 2024
+last_updated: Sep 28, 2024
 ---
 
-It's impossible to mitigate `LockCouldNotBeAcquiredExecption` exceptions on the infrastructure level, for example, by increasing CPU or memory resources. To make sure an application reaches expected scaling targets, during performance and load testing, take into account the scaling challenges described in this document.
+A session lock is necessary to prevent multiple processes belonging to a single user's session from accessing or modifying session data at the same time, ensuring data integrity and consistency. 
+It helps avoid race conditions and data corruption in high-concurrency environments.
+While a user's session is being accessed by a process, other processes of the same user's session can't access it; the session is *locked*.
 
+The handler is configured with:
+
+```php
+// config/Shared/config_default.php
+$config[SessionConstants::YVES_SESSION_SAVE_HANDLER] = SessionRedisConfig::SESSION_HANDLER_REDIS_LOCKING;
+```
+If a process can't lock the session, it tries to acquire it by using a *spinlock* strategy. 
+It waits for 0.01 seconds by default and then retries acquiring a lock to the session. The retry timeout can be configured:
+```php
+// config/Shared/config_default.php
+$config[SessionRedisConstants::LOCKING_RETRY_DELAY_MICROSECONDS] = 0;
+```
+
+These attempts continue until the timeout, which is 0.8 from `max_execution_time` by default. The timeout also can be configured:
```

Comment:
- here, I'd use "80% of " instead of "0.8 from" - easier to "digest"
- and we have to add that it will only apply when nothing is configured here in config or when the value is 0, like in this example, otherwise - it'll use configured value

Reply:
done